lab3/client/src/reducer.js

```diff
@@ -0,0 +1,44 @@
+import constants from './constants';
+import Generator from '../src/models/generator';
+export default function reducer (state, action) {
+	
+	switch (action.type) {
+	case 'EXAMPLE_MUTATION':
+		state.example = action.payload;
+		return state;
+
+	case constants.actions.BUTTON_CLICK:
+		state.counter++;
+		return state;
+		
+	case 'BUY_GENERATOR':
+ 
+	for (var i = 0; i < state.generators.length; i++)
+
+	 {
+		 if (state.generators[i].name===action.payload.name)
+		 {
+			var index = i;
+		 }
+		}
+		const generator = state.generators[index];
+	if(state.counter >= generator.baseCost)
+	{  
+		 console.log(this,generator);// debugging
+	   	 state.counter -= generator.baseCost;
```

Comment:
need to use `getCost` function here.

Reply:
yea i saw that by hw1,  since the button was not increasing  the cost  , instead was keeping it at the same cost at the time